File: js/task3a/S3_OBJ_02.js
function mergeDefaults(defaults, overrides) {
  return {
    ...(defaults ?? {}),
    ...(overrides ?? {})
  };
}

const defaults = { a: 1, b: 2 };
const overrides = { b: 3, c: 4 };

const result = mergeDefaults(defaults, overrides);

console.log("Result:", result);
console.log("Defaults unchanged:", defaults); 
console.log("Overrides unchanged:", overrides); 
console.log(mergeDefaults({ a: 1 }, null));       
console.log(mergeDefaults(null, { b: 2 }));       
console.log(mergeDefaults(undefined, undefined)); 

File: python/task3a/S3_OBJ_02.py
def mergeDefaults(defaults, overrides):
  defaults = defaults or {}
  overrides = overrides or {}
  return {**defaults, **overrides}


defaults = {"a": 1, "b": 2}
overrides = {"b": 3, "c": 4}

result = mergeDefaults(defaults, overrides)

print("Result:", result)
print("Defaults unchanged:", defaults)
print("Overrides unchanged:", overrides)
print(mergeDefaults({"a": 1}, None))
print(mergeDefaults(None, {"b": 2}))
print(mergeDefaults(None, None))
